core/dictgen/src/rootcling_impl.cxx

```diff
@@ -207,6 +215,19 @@ const char *GetExePath()
       exepath = buf;
     }
 #endif
+#if defined R__FBSD || defined __FreeBSD__
+  procstat* ps = procstat_open_sysctl();  //
+  kinfo_proc* kp = kinfo_getproc(getpid());
```

Comment:
I saw some other usages of the procfs filesystem typically mounted on `/proc` on Linux, namely in:
- The `proof/`, `rootx/` and `bindings/pyroot/cppyy/` directories
- The `config/thisroot.sh` file

Maybe we should take a look into those too.

Reply:
config/thisroot.sh should work as it is
rootx I'll fix, but is there no common library/place to define GetExePath (and the like) once for all components?
proof and python I'll leave for later as I have no use for those. Someone will contribute a patch if there is need:)